docs/api/QUIC_STREAM_EVENT.md

```diff
@@ -0,0 +1,266 @@
+
+## QUIC_STREAM_EVENT_SEND_COMPLETE
+
+Indicates that MsQuic has completed a [StreamSend](StreamSend.md) operation initated by the application.
+
+### SEND_COMPLETE
+
+Data for `StreamSend` completion is included in this struct/union.
+
+`Canceled`
+Indicates that the StreamSend operation was canceled.
+
+`ClientContext`
+Client context to match this event with the original `StreamSend` operation.
+
+## QUIC_STREAM_EVENT_PEER_SEND_SHUTDOWN
+
+Indicates that the current send operation to the peer on this stream has been shutdown/completed.
+
+## QUIC_STREAM_EVENT_PEER_SEND_ABORTED
+
+Indicates that the peer has aborted `StreamSend` operation.
+
+### SEND_ABORTED
+
+Additional details of the send abort event are passed in this struct/union.
+
+`ErrorCode`
+
+Application's protocol specific, 62-bit error code.
+
+## QUIC_STREAM_EVENT_PEER_RECEIVE_ABORTED
+
+Indicates that the peer has aborted receiving data.
+
+### RECEIVE_ABORTED
+
+Additional details of the receive abort event are passed in this struct/union.
+
+`ErrorCode`
+
+Application's protocol specific, 62-bit error code.
+
+## QUIC_STREAM_EVENT_SEND_SHUTDOWN_COMPLETE
+
+This event is raised when a stream send direction has been fully shut down.
+
+### SEND_SHUTDOWN_COMPLETE
+
+Additional details of send shutdown completion are passed in this struct/union.
+
+`Graceful`
+
+TRUE if the send shutdown operation was gracefully shutdown, FALSE otherwise.
+
+## QUIC_STREAM_EVENT_SHUTDOWN_COMPLETE
+
+This event indicates that the stream has been completely shutdown.
+
+### SHUTDOWN_COMPLETE
+
+Additional details for the stream shutdown are passed in this struct/union.
+
+`ConnectionShutdown`
+
+If TRUE, the Connection corresponding to this stream has been closed locally or remotely.
+
+`AppCloseInProgress`
+
+If TRUE, the application is in the process of closing the stream.
+
+`ConnectionShutdownByApp`
+
+If TRUE, the application shutdown the Connection corresponding to this stream.
+
+`ConnectionClosedRemotely`
+
+If TRUE, the Connection corresponding to this stream has been closed remotely.
+
+`ConnectionErrorCode`
+
+62-bit Connection closure error code, if any.
+
+`ConnectionCloseStatus`
+
+QUIC_STATUS value of the connection close operation, if any.
+
+## QUIC_STREAM_EVENT_IDEAL_SEND_BUFFER_SIZE
+
+MsQuic indicates the ideal send buffer size to the application through this event, so as not to idle the connection.
+
+### IDEAL_SEND_BUFFER_SIZE
+
+Ideal send buffer size is indicated in this struct/union.
+
+`ByteCount`
+
+Ideal send buffer size in bytes for each `StreamSend` operation to avoid idling the connection.
+
+## QUIC_STREAM_EVENT_PEER_ACCEPTED
+
+This event is raised when a peer has provided sufficient flow control to accept a new stream. See [StreamStart](StreamStart.md) for additional information.
+
+## QUIC_STREAM_EVENT_CANCEL_ON_LOSS
+
+This event is raised when a stream is shutdown due to packet loss
```

Comment:
nit: Would be nice to link to the `QUIC_SEND_FLAG_CANCEL_ON_LOSS` option, since that behavior is not the default.